crml/transaction-payment/src/lib.rs

```diff
@@ -176,7 +191,12 @@ where
 		len: usize,
 	) -> TransactionValidity {
 		// pay any fees.
-		let fee = Self::compute_fee(len, info, self.0);
+		let fee = Self::compute_fee(len, info, self.tip);
+		if let Some(fee_exchange) = &self.fee_exchange {
+			if let Err(_) = T::BuyFeeAsset::buy_fee_asset(who, fee, fee_exchange) {
```

Comment:
clippy would say to use `if T::BuyFeeAsset::buy_fee_asset(who, fee, fee_exchange).is_err() {` but TIOLI